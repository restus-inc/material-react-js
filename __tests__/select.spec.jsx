/* global jest, describe, it, expect */
import React from 'react';
import renderer from 'react-test-renderer';

import Select from '../lib/select';

jest.mock('@material/select');

describe('Select component', () => {
  it('supports filled select', () => {
    let component = renderer.create(
      <Select items={['one', 'two', 'three']}/>,
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      <Select label="foo" items={['one', 'two', 'three']} required={true} disabled={true}/>,
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      <Select label="foo" id="bar" items={['one', 'two', 'three']} value="baz" className="qux"/>,
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      <Select variation="filled" label="foo" items={['one', 'two', 'three']} required={true} disabled={true}/>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('supports outlined select', () => {
    let component = renderer.create(
      <Select variation="outlined" items={['one', 'two', 'three']}/>,
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      <Select variation="outlined" label="foo" items={['one', 'two', 'three']} required={true} disabled={true}/>,
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      <Select variation="outlined" label="foo" id="bar" items={['one', 'two', 'three']} value="baz" className="qux"/>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('allow to set objects as items', () => {
    const items = [
      { key: '1', value: 'one' },
      { key: '2', value: 'two' },
      { key: '3', value: 'three' },
    ];
    let component = renderer.create(
      <Select label="foo" items={items} value="1" itemsValueAttr="key" itemsTextAttr="value" />,
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      <Select label="foo" items={items} value="0" itemsTextAttr="value" />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
