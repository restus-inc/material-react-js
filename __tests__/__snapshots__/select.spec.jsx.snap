// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component allow to set objects as items 1`] = `
"<div
  class=\\"mdc-select mdc-select--filled\\"
>
  <div
    aria-disabled=\\"false\\"
    aria-haspopup=\\"listbox\\"
    class=\\"mdc-select__anchor\\"
    role=\\"button\\"
    tabindex=\\"0\\"
  >
    <span
      class=\\"mdc-select__ripple\\"
    />
    <span
      class=\\"mdc-select__selected-text\\"
    >
      two
    </span>
    <span
      class=\\"mdc-select__dropdown-icon\\"
    >
      <svg
        class=\\"mdc-select__dropdown-icon-graphic\\"
        viewBox=\\"7 10 10 5\\"
      >
        <polygon
          class=\\"mdc-select__dropdown-icon-inactive\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 10 12 15 17 10\\"
          stroke=\\"none\\"
        />
        <polygon
          class=\\"mdc-select__dropdown-icon-active\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 15 12 10 17 15\\"
          stroke=\\"none\\"
        />
      </svg>
    </span>
    <span
      class=\\"mdc-floating-label mdc-floating-label--float-above\\"
    >
      foo
    </span>
    <span
      class=\\"mdc-line-ripple\\"
    />
  </div>
  <div
    class=\\"mdc-select__menu mdc-menu mdc-menu-surface mdc-menu-surface--fullwidth\\"
  >
    <ul
      class=\\"mdc-list\\"
      role=\\"listbox\\"
    >
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"1\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          one
        </span>
      </li>
      <li
        aria-selected=\\"true\\"
        class=\\"mdc-list-item mdc-list-item--selected\\"
        data-value=\\"2\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          two
        </span>
      </li>
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"3\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          three
        </span>
      </li>
    </ul>
  </div>
</div>"
`;

exports[`Select component allow to set objects as items 2`] = `
"<div
  class=\\"mdc-select mdc-select--filled\\"
>
  <div
    aria-disabled=\\"false\\"
    aria-haspopup=\\"listbox\\"
    class=\\"mdc-select__anchor\\"
    role=\\"button\\"
    tabindex=\\"0\\"
  >
    <span
      class=\\"mdc-select__ripple\\"
    />
    <span
      class=\\"mdc-select__selected-text\\"
    >
      two
    </span>
    <span
      class=\\"mdc-select__dropdown-icon\\"
    >
      <svg
        class=\\"mdc-select__dropdown-icon-graphic\\"
        viewBox=\\"7 10 10 5\\"
      >
        <polygon
          class=\\"mdc-select__dropdown-icon-inactive\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 10 12 15 17 10\\"
          stroke=\\"none\\"
        />
        <polygon
          class=\\"mdc-select__dropdown-icon-active\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 15 12 10 17 15\\"
          stroke=\\"none\\"
        />
      </svg>
    </span>
    <span
      class=\\"mdc-floating-label mdc-floating-label--float-above\\"
    >
      foo
    </span>
    <span
      class=\\"mdc-line-ripple\\"
    />
  </div>
  <div
    class=\\"mdc-select__menu mdc-menu mdc-menu-surface mdc-menu-surface--fullwidth\\"
  >
    <ul
      class=\\"mdc-list\\"
      role=\\"listbox\\"
    >
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"0\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          one
        </span>
      </li>
      <li
        aria-selected=\\"true\\"
        class=\\"mdc-list-item mdc-list-item--selected\\"
        data-value=\\"1\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          two
        </span>
      </li>
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"2\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          three
        </span>
      </li>
    </ul>
  </div>
</div>"
`;

exports[`Select component supports filled select 1`] = `
"<div
  class=\\"mdc-select mdc-select--filled\\"
>
  <div
    aria-disabled=\\"false\\"
    aria-haspopup=\\"listbox\\"
    class=\\"mdc-select__anchor\\"
    role=\\"button\\"
    tabindex=\\"0\\"
  >
    <span
      class=\\"mdc-select__ripple\\"
    />
    <span
      class=\\"mdc-select__selected-text\\"
    />
    <span
      class=\\"mdc-select__dropdown-icon\\"
    >
      <svg
        class=\\"mdc-select__dropdown-icon-graphic\\"
        viewBox=\\"7 10 10 5\\"
      >
        <polygon
          class=\\"mdc-select__dropdown-icon-inactive\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 10 12 15 17 10\\"
          stroke=\\"none\\"
        />
        <polygon
          class=\\"mdc-select__dropdown-icon-active\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 15 12 10 17 15\\"
          stroke=\\"none\\"
        />
      </svg>
    </span>
    <span
      class=\\"mdc-line-ripple\\"
    />
  </div>
  <div
    class=\\"mdc-select__menu mdc-menu mdc-menu-surface mdc-menu-surface--fullwidth\\"
  >
    <ul
      class=\\"mdc-list\\"
      role=\\"listbox\\"
    >
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"one\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          one
        </span>
      </li>
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"two\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          two
        </span>
      </li>
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"three\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          three
        </span>
      </li>
    </ul>
  </div>
</div>"
`;

exports[`Select component supports filled select 2`] = `
"<div
  class=\\"mdc-select mdc-select--filled mdc-select--required mdc-select--disabled\\"
>
  <div
    aria-disabled=\\"true\\"
    aria-haspopup=\\"listbox\\"
    aria-invalid=\\"false\\"
    aria-required=\\"true\\"
    class=\\"mdc-select__anchor\\"
    role=\\"button\\"
  >
    <span
      class=\\"mdc-select__ripple\\"
    />
    <span
      class=\\"mdc-select__selected-text\\"
    />
    <span
      class=\\"mdc-select__dropdown-icon\\"
    >
      <svg
        class=\\"mdc-select__dropdown-icon-graphic\\"
        viewBox=\\"7 10 10 5\\"
      >
        <polygon
          class=\\"mdc-select__dropdown-icon-inactive\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 10 12 15 17 10\\"
          stroke=\\"none\\"
        />
        <polygon
          class=\\"mdc-select__dropdown-icon-active\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 15 12 10 17 15\\"
          stroke=\\"none\\"
        />
      </svg>
    </span>
    <span
      class=\\"mdc-floating-label mdc-floating-label--required\\"
    >
      foo
    </span>
    <span
      class=\\"mdc-line-ripple\\"
    />
  </div>
  <div
    class=\\"mdc-select__menu mdc-menu mdc-menu-surface mdc-menu-surface--fullwidth\\"
  >
    <ul
      class=\\"mdc-list\\"
      role=\\"listbox\\"
    >
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"one\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          one
        </span>
      </li>
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"two\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          two
        </span>
      </li>
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"three\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          three
        </span>
      </li>
    </ul>
  </div>
</div>"
`;

exports[`Select component supports filled select 3`] = `
"<div
  class=\\"mdc-select mdc-select--filled qux\\"
>
  <div
    aria-disabled=\\"false\\"
    aria-haspopup=\\"listbox\\"
    class=\\"mdc-select__anchor\\"
    role=\\"button\\"
    tabindex=\\"0\\"
  >
    <span
      class=\\"mdc-select__ripple\\"
    />
    <span
      class=\\"mdc-select__selected-text\\"
    >
      two
    </span>
    <span
      class=\\"mdc-select__dropdown-icon\\"
    >
      <svg
        class=\\"mdc-select__dropdown-icon-graphic\\"
        viewBox=\\"7 10 10 5\\"
      >
        <polygon
          class=\\"mdc-select__dropdown-icon-inactive\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 10 12 15 17 10\\"
          stroke=\\"none\\"
        />
        <polygon
          class=\\"mdc-select__dropdown-icon-active\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 15 12 10 17 15\\"
          stroke=\\"none\\"
        />
      </svg>
    </span>
    <span
      class=\\"mdc-floating-label mdc-floating-label--float-above\\"
    >
      foo
    </span>
    <span
      class=\\"mdc-line-ripple\\"
    />
  </div>
  <div
    class=\\"mdc-select__menu mdc-menu mdc-menu-surface mdc-menu-surface--fullwidth\\"
  >
    <ul
      class=\\"mdc-list\\"
      role=\\"listbox\\"
    >
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"one\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          one
        </span>
      </li>
      <li
        aria-selected=\\"true\\"
        class=\\"mdc-list-item mdc-list-item--selected\\"
        data-value=\\"two\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          two
        </span>
      </li>
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"three\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          three
        </span>
      </li>
    </ul>
  </div>
</div>"
`;

exports[`Select component supports filled select 4`] = `
"<div
  class=\\"mdc-select mdc-select--filled mdc-select--required mdc-select--disabled\\"
>
  <div
    aria-disabled=\\"true\\"
    aria-haspopup=\\"listbox\\"
    aria-invalid=\\"false\\"
    aria-required=\\"true\\"
    class=\\"mdc-select__anchor\\"
    role=\\"button\\"
  >
    <span
      class=\\"mdc-select__ripple\\"
    />
    <span
      class=\\"mdc-select__selected-text\\"
    />
    <span
      class=\\"mdc-select__dropdown-icon\\"
    >
      <svg
        class=\\"mdc-select__dropdown-icon-graphic\\"
        viewBox=\\"7 10 10 5\\"
      >
        <polygon
          class=\\"mdc-select__dropdown-icon-inactive\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 10 12 15 17 10\\"
          stroke=\\"none\\"
        />
        <polygon
          class=\\"mdc-select__dropdown-icon-active\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 15 12 10 17 15\\"
          stroke=\\"none\\"
        />
      </svg>
    </span>
    <span
      class=\\"mdc-floating-label mdc-floating-label--required\\"
    >
      foo
    </span>
    <span
      class=\\"mdc-line-ripple\\"
    />
  </div>
  <div
    class=\\"mdc-select__menu mdc-menu mdc-menu-surface mdc-menu-surface--fullwidth\\"
  >
    <ul
      class=\\"mdc-list\\"
      role=\\"listbox\\"
    >
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"one\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          one
        </span>
      </li>
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"two\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          two
        </span>
      </li>
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"three\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          three
        </span>
      </li>
    </ul>
  </div>
</div>"
`;

exports[`Select component supports outlined select 1`] = `
"<div
  class=\\"mdc-select mdc-select--outlined\\"
>
  <div
    aria-disabled=\\"false\\"
    aria-haspopup=\\"listbox\\"
    class=\\"mdc-select__anchor\\"
    role=\\"button\\"
    tabindex=\\"0\\"
  >
    
    <span
      class=\\"mdc-select__selected-text\\"
    />
    <span
      class=\\"mdc-select__dropdown-icon\\"
    >
      <svg
        class=\\"mdc-select__dropdown-icon-graphic\\"
        viewBox=\\"7 10 10 5\\"
      >
        <polygon
          class=\\"mdc-select__dropdown-icon-inactive\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 10 12 15 17 10\\"
          stroke=\\"none\\"
        />
        <polygon
          class=\\"mdc-select__dropdown-icon-active\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 15 12 10 17 15\\"
          stroke=\\"none\\"
        />
      </svg>
    </span>
    <span
      class=\\"mdc-notched-outline mdc-notched-outline--no-label\\"
    >
      <span
        class=\\"mdc-notched-outline__leading\\"
      />
      <span
        class=\\"mdc-notched-outline__trailing\\"
      />
    </span>
  </div>
  <div
    class=\\"mdc-select__menu mdc-menu mdc-menu-surface mdc-menu-surface--fullwidth\\"
  >
    <ul
      class=\\"mdc-list\\"
      role=\\"listbox\\"
    >
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"one\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          one
        </span>
      </li>
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"two\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          two
        </span>
      </li>
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"three\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          three
        </span>
      </li>
    </ul>
  </div>
</div>"
`;

exports[`Select component supports outlined select 2`] = `
"<div
  class=\\"mdc-select mdc-select--outlined mdc-select--required mdc-select--disabled\\"
>
  <div
    aria-disabled=\\"true\\"
    aria-haspopup=\\"listbox\\"
    aria-invalid=\\"false\\"
    aria-required=\\"true\\"
    class=\\"mdc-select__anchor\\"
    role=\\"button\\"
  >
    
    <span
      class=\\"mdc-select__selected-text\\"
    />
    <span
      class=\\"mdc-select__dropdown-icon\\"
    >
      <svg
        class=\\"mdc-select__dropdown-icon-graphic\\"
        viewBox=\\"7 10 10 5\\"
      >
        <polygon
          class=\\"mdc-select__dropdown-icon-inactive\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 10 12 15 17 10\\"
          stroke=\\"none\\"
        />
        <polygon
          class=\\"mdc-select__dropdown-icon-active\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 15 12 10 17 15\\"
          stroke=\\"none\\"
        />
      </svg>
    </span>
    <span
      class=\\"mdc-notched-outline mdc-notched-outline--upgraded\\"
    >
      <span
        class=\\"mdc-notched-outline__leading\\"
      />
      <span
        class=\\"mdc-notched-outline__notch\\"
      >
        <span
          class=\\"mdc-floating-label mdc-floating-label--required\\"
          style=\\"transition-duration: 0s;\\"
        >
          foo
        </span>
      </span>
      <span
        class=\\"mdc-notched-outline__trailing\\"
      />
    </span>
  </div>
  <div
    class=\\"mdc-select__menu mdc-menu mdc-menu-surface mdc-menu-surface--fullwidth\\"
  >
    <ul
      class=\\"mdc-list\\"
      role=\\"listbox\\"
    >
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"one\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          one
        </span>
      </li>
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"two\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          two
        </span>
      </li>
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"three\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          three
        </span>
      </li>
    </ul>
  </div>
</div>"
`;

exports[`Select component supports outlined select 3`] = `
"<div
  class=\\"mdc-select mdc-select--outlined qux\\"
>
  <div
    aria-disabled=\\"false\\"
    aria-haspopup=\\"listbox\\"
    class=\\"mdc-select__anchor\\"
    role=\\"button\\"
    tabindex=\\"0\\"
  >
    
    <span
      class=\\"mdc-select__selected-text\\"
    >
      two
    </span>
    <span
      class=\\"mdc-select__dropdown-icon\\"
    >
      <svg
        class=\\"mdc-select__dropdown-icon-graphic\\"
        viewBox=\\"7 10 10 5\\"
      >
        <polygon
          class=\\"mdc-select__dropdown-icon-inactive\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 10 12 15 17 10\\"
          stroke=\\"none\\"
        />
        <polygon
          class=\\"mdc-select__dropdown-icon-active\\"
          fill-rule=\\"evenodd\\"
          points=\\"7 15 12 10 17 15\\"
          stroke=\\"none\\"
        />
      </svg>
    </span>
    <span
      class=\\"mdc-notched-outline mdc-notched-outline--upgraded mdc-notched-outline--notched\\"
    >
      <span
        class=\\"mdc-notched-outline__leading\\"
      />
      <span
        class=\\"mdc-notched-outline__notch\\"
        style=\\"width: 0px;\\"
      >
        <span
          class=\\"mdc-floating-label mdc-floating-label--float-above\\"
          style=\\"transition-duration: 0s;\\"
        >
          foo
        </span>
      </span>
      <span
        class=\\"mdc-notched-outline__trailing\\"
      />
    </span>
  </div>
  <div
    class=\\"mdc-select__menu mdc-menu mdc-menu-surface mdc-menu-surface--fullwidth\\"
  >
    <ul
      class=\\"mdc-list\\"
      role=\\"listbox\\"
    >
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"one\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          one
        </span>
      </li>
      <li
        aria-selected=\\"true\\"
        class=\\"mdc-list-item mdc-list-item--selected\\"
        data-value=\\"two\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          two
        </span>
      </li>
      <li
        class=\\"mdc-list-item\\"
        data-value=\\"three\\"
        role=\\"option\\"
        tabindex=\\"-1\\"
      >
        <span
          class=\\"mdc-list-item__ripple\\"
        />
        <span
          class=\\"mdc-list-item__text\\"
        >
          three
        </span>
      </li>
    </ul>
  </div>
</div>"
`;
